platform/platform.cpp

```diff
@@ -135,11 +145,24 @@ string Platform::ReadPathForFile(string const & file, string searchScope) const
   {
     switch (searchScope[i])
     {
-    case 'w': fullPath = m_writableDir + file; break;
-    case 'r': fullPath = m_resourcesDir + file; break;
-    case 's': fullPath = m_settingsDir + file; break;
-    case 'f': fullPath = file; break;
-    default : CHECK(false, ("Unsupported searchScope:", searchScope)); break;
+    case 'w':
+      ASSERT(!m_writableDir.empty(), ("ReadPathForFile m_writableDir is empty"));
```

Comment:
nit: ASSERT macro will print function's name and passed expression "!m_writableDir.empty()", so additional string literal comment is useless here and below.